scripts/usb_reset.py

```diff
@@ -212,17 +228,36 @@ def setup_cgroup(domid, pid):
 
     try:
         # unbuffered write to ensure each one is flushed immediately
-        with open(cg_dir + "/tasks", "w", 0) as tasks, \
-                open(cg_dir + "/devices.deny", "w", 0) as deny, \
-                open(cg_dir + "/devices.allow", "w", 0) as allow:
+        # to the kernel's control group filesystem:
+        #
+        # The order of writes is likely not important, but the writes
+        # may have to be a single write() system call for the entire string.
+        #
+        # Using the unbuffered Raw IO mode, we know the write was done
+        # in exactly this way by the write function call itself, not later.
+        #
+        # With such short writes, splitting the writes because of a
+        # too small buffer will not usually happen, but be paranoid
+        # and keep the same code for Python3 too. We've to use binary
+        # mode for Python3 for this as Python3 does now support unbuffered
+        # Raw I/O in Text mode (with implicit unicode str to byte encode()).
```

Comment:
I wrote this comment in a bit of a hurry.
When reading it now, I found that its wording was not great.
I propose to rephrase it like in this proposed commit/change:
```suggestion
        # With small writes like thise, splitting them because of overflowing the
        # buffer is not expected to happen. But stay on safe and keep using unbuffered IO.
        # We've to migrate to binary mode for Python3 for this because
        # Python3 only supports unbuffered raw I/O in binary mode:
```

Reply:
Done